nixos/modules/programs/foliate.nix

```diff
@@ -0,0 +1,117 @@
+{ lib, pkgs, config, ... }:
+
+with lib; {
+  options.programs.foliate = {
+    enable = mkEnableOption "Foliate - A simple and modern eBook reader";
+
+    footer = {
+      left = mkOption {
+        description = "Information to show on the left footer";
+        default = "location";
+        type = types.enum [
+          "location"
+          "percentage"
+          "time-left-book"
+          "time-left-section"
+          "section-name"
+        ];
+      };
+      right = mkOption {
+        description = "Information to show on the right footer";
+        default = "section-name";
+        type = types.enum [
+          "location"
+          "percentage"
+          "time-left-book"
+          "time-left-section"
+          "section-name"
+        ];
+      };
+    };
+
+    layout = mkOption {
+      description = "Layout between pages of the eBook";
+      default = "continuous";
+      type = types.enum [ "auto" "scrolled" "continuous" "single" ];
+    };
+
+    margin = mkOption {
+      description = "Distance between the window border and the eBook contents";
+      default = 0;
+      type = types.ints.unsigned;
+    };
+
+    spacing = mkOption {
+      description = "Spacing between characters";
+      default = 1.0;
+      type = types.floats.between 1.0 3.0;
+    };
+
+    theme = {
+      bgColor = mkOption {
+        description = "Background color";
+        default = "#fff";
+        example = "#000";
+        type = types.str;
+      };
+
+      font = mkOption {
+        description = "Font to use";
+        default = null;
+        example = "Fira Code 16";
+        type = types.nullOr types.str;
+      };
+
+      fgColor = mkOption {
+        description = "Foreground color";
+        default = "#000";
+        example = "#fff";
+        type = types.str;
+      };
+
+      linkColor = mkOption {
+        description = "Link color";
+        default = "blue";
+        example = "#6495ed";
+        type = types.str;
+      };
+    };
+  };
+  config = lib.mkIf config.programs.foliate.enable {
+    environment.systemPackages = [ pkgs.foliate ];
+
+    programs.dconf.enable = true;
```

Comment:
dconf is deprecated. Can't we use something else?

Reply:
Sadly this is what the ebook reader uses for configuration, there is not much we can do here, there are no alternative configuration interfaces supported by the program

Reply:
oh well...